### ISSUE TITLE
`fatal: --git-path requires an argument`

### ISSUE THREAD

**Issue by PotcFdk** [qgadrian/elixir_git_hooks#191]
After installing, or when running `mix git_hooks.install`, I get multiple `fatal: --git-path requires an argument`.
I believe this is because there are cases that use some git commands incorrectly.

One example that I found seems to be here:
https://github.com/qgadrian/elixir_git_hooks/blob/79b9aec14dbec5f0c24d0d2dfdadc102799c1ab0/lib/git/path.ex#L23-L24
This calls `resolve_git_path_based_on_git_version` without any arguments.
That function then seems to default to `""` and hence calls git rev-parse with the relevant argument missing:
https://github.com/qgadrian/elixir_git_hooks/blob/79b9aec14dbec5f0c24d0d2dfdadc102799c1ab0/lib/git/path.ex#L56-L70

Let's see what rev-parse does with `--git-path` but with no further arguments:
```shell
$ git rev-parse --git-path       
fatal: --git-path requires an argument
```
Aha. That same function, `resolve_git_path_based_on_git_version`, seems to also have a case that uses `--git-dir`. Let's see what that one does:
```shell
$ git rev-parse --git-dir 
.git
```
No errors there. Maybe the expected git rev-parse behavior changed?

Tested under Windows 11.
```shell
$ git version
git version 2.46.0.windows.1
```

**Comment by qgadrian** [comment]
Update on this, I published a new `v0.8.0-pre1` release that hopefully resolves this issue. 

I will wait to get some feedback before publishing a production release.

**Comment by qgadrian** [comment]
Released a new version [`v0.8.0`](https://github.com/qgadrian/elixir_git_hooks/releases/tag/v0.8.0) that I hope fixes this issue.
